mypy/checkexpr.py

```diff
@@ -3146,11 +3147,10 @@ def combine_function_signatures(self, types: list[ProperType]) -> AnyType | Call
         # confusing and ought to be re-written anyways.)
         callables, variables = merge_typevars_in_callables_by_name(callables)
 
-        new_args: list[list[Type]] = [[] for _ in range(len(callables[0].arg_types))]
-        new_kinds = list(callables[0].arg_kinds)
-        new_returns: list[Type] = []
-
+        new_args: list[list[Type]] = [[] for _ in callables[0].arg_types]
+        new_kinds, new_returns = list(callables[0].arg_kinds), []
```

Comment:
This looks like a penalty to readability rather then refactoring, why merge two unrelated assignments?